public/app/features/plugins/sql/datasource/SqlDatasource.ts

```diff
@@ -115,6 +125,50 @@ export abstract class SqlDatasource extends DataSourceWithBackend<SQLQuery, SQLO
     };
   }
 
+  query(request: DataQueryRequest<SQLQuery>): Observable<DataQueryResponse> {
+    // This logic reenables the previous SQL behavior regarding what databases are available for the user to query.
+    if (isSqlDatasourceDatabaseSelectionFeatureFlagEnabled()) {
+      const databaseIssue = this.checkForDatabaseIssue(request);
+
+      if (!!databaseIssue) {
+        const error = new Error(databaseIssue);
+        return throwError(() => error);
+      }
+    }
+
+    return super.query(request);
+  }
+
+  private checkForDatabaseIssue(request: DataQueryRequest<SQLQuery>) {
+    // If the datasource is Postgres and there is no default database configured - either never configured or removed - return a database issue.
+    if (this.type === 'postgres' && !this.preconfiguredDatabase) {
+      return `You do not currently have a default database configured for this data source. Postgres requires a default
+             database with which to connect. Please configure one through the Data Sources Configuration page, or if you
+             are using a provisioning file, update that configuration file with a default database.`;
+    }
+
+    // No need to check for database change/update issues if the datasource is being used in Explore.
+    if (request.app !== 'explore') {
```

Comment:
I think this could be replaced with `CoreApp` enum from `@grafana/data`

Reply:
Updated.